File: jutge/Funcions/number-of-digits.py

def number_of_digits(n:int) -> int:
    """Retorna el nombre de digits d'un nombre n"""
    if n == 0:
        return 1
    else:
        d = 0
        while n != 0:
            d += 1
            n //= 10
    return d

File: jutge/cpp/number-of-digits.cpp
#include <iostream>
using namespace std;

int numberOfDigits(int n) {
  if (n < 10) {
    return 1;
  } else {
    return 1 + numberOfDigits(n / 10);
  }
}

int main() {
  int n;
  cin >> n;
  cout << "The number of digits of " << n << " is " << numberOfDigits(n) << "."
       << endl;
}
